packages/web-fetch/src/index.ts

```diff
@@ -0,0 +1,96 @@
+import {reatomAsync} from '@reatom/async'
+import {Atom, Ctx, isAtom, throwReatomError} from '@reatom/core'
+
+export interface ReatomFetchOptions<O, I> {
+	fetch?: (url: string, init: RequestInit) => Promise<Response>
+	init?: Omit<RequestInit, 'method' | 'signal' | 'body' | 'headers'>
+	auth?: MaybeAtom<string>
+	headers?: MaybeAtom<HeadersInit>
+	baseUrl?: string
+	method?: string
+	validate?: (o: O) => boolean | string
+	parseResponse?: (response: Response) => Promise<O>
+	serializeBody?: (i: I, headers: Headers) => BodyInit
+	validateStatus?: (status: number) => boolean | string
+}
+
+type UrlSearchParamsInit = ConstructorParameters<typeof URLSearchParams>[0]
+
+type MaybeAtom<T> = Atom<T> | T
+
+const defaults: ReatomFetchOptions<unknown, unknown> = {
+	init: {},
+	fetch,
+	validate: () => true,
+	validateStatus: status => status >= 200 && status < 300,
+	parseResponse: response => {
+		const ct = response.headers.get(`content-type`)
+		if (ct !== `application/json`) {
+			throw new Error(`Invalid response content-type "${ct}"`)
+		}
+		return response.json()
+	},
+	serializeBody: body => JSON.stringify(body),
+}
+
+export const createReatomFetch = (
+	options: ReatomFetchOptions<unknown, unknown>
+) => {
+	const baseOpts = {...defaults, ...options}
+
+	return <O, I = void>(url: string, opts: ReatomFetchOptions<O, I> = {}) =>
+		reatomAsync(async (ctx, i: I, params?: UrlSearchParamsInit) => {
+			const mergedOpts = {...baseOpts, ...opts}
+
+			const baseHeaders = [...new Headers(unwrap(ctx, baseOpts.headers))]
+			const headerOverrides = new Headers(unwrap(ctx, opts.headers))
+
+			const headers = new Headers([
+				...new Headers(baseHeaders.filter(x => !headerOverrides.has(x[0]))),
+				...headerOverrides,
+			])
+
+			const auth = unwrap(ctx, mergedOpts.auth)
+			if (auth) headers.set(`authorization`, auth)
+
+			const base = mergedOpts.baseUrl
+			if (base) {
+				url = url.replace(
+					/^(?!.*\/\/)\/?/,
+					base.endsWith('/') ? base : base + '/'
+				)
+			}
+
+			if (params) url += `?` + new URLSearchParams(params).toString()
+
+			const resp = await mergedOpts.fetch!(url, {
+				...mergedOpts.init,
+				method: mergedOpts.method,
+				body: i != null ? mergedOpts.serializeBody!(i, headers) : undefined,
+				signal: ctx.controller.signal,
+				headers,
+			})
```

Comment:
Add `NetworkError` class pls.
```suggestion
			}).catch((e) => new NetworkError(e))
```